cpp/parallel/CMakeLists.txt

```diff
@@ -1,3 +1,15 @@
-add_library(_parallel OBJECT tbb_config.h tbb_config.cc)
+nanobind_add_module(_parallel NB_SHARED module-parallel.cc)
+target_link_libraries(_parallel PUBLIC libfreud)
 
-target_link_libraries(_parallel PUBLIC TBB::tbb)
+if(APPLE)
+  set_target_properties(_parallel PROPERTIES INSTALL_RPATH "@loader_path")
+else()
+  set_target_properties(_parallel PROPERTIES INSTALL_RPATH "\$ORIGIN")
+endif()
+
+if(_using_conda OR DEFINED ENV{CIBUILDWHEEL})
+  set_target_properties(_parallel PROPERTIES INSTALL_RPATH_USE_LINK_PATH True)
+endif()
```

Comment:
Will need to move this to a CMake macro or function to avoid code duplication as we add more modules.

Reply:
Yes, I can do this on the next PR for the parallel module